packages/dropdowns.next/src/elements/menu/Menu.tsx

```diff
@@ -5,42 +5,103 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import React, { forwardRef, useContext, useMemo } from 'react';
+import React, { RefObject, forwardRef, useContext, useMemo, useRef } from 'react';
 import PropTypes from 'prop-types';
-import { ThemeContext } from 'styled-components';
-import { DEFAULT_THEME } from '@zendeskgarden/react-theming';
-import { IMenuProps, PLACEMENT } from '../../types';
+import mergeRefs from 'react-merge-refs';
+import { IMenuProps, IMenuButtonProps, PLACEMENT } from '../../types';
 import { MenuContext } from '../../context/useMenuContext';
-import { StyledFloatingMenu, StyledMenu } from '../../views';
-import { toArrowPosition, toFloatingPlacement, toMenuPosition } from './utils';
+import { MenuList } from './MenuList';
+import { StyledButton } from '../../views';
+import { useMenu } from '@zendeskgarden/container-menu';
+import { toItems } from './utils';
+import { ThemeContext } from 'styled-components';
+import { DEFAULT_THEME, useWindow } from '@zendeskgarden/react-theming';
 
 /**
  * @extends HTMLAttributes<HTMLUListElement>
  */
 export const Menu = forwardRef<HTMLUListElement, IMenuProps>(
-  ({ children, hasArrow, isCompact, maxHeight, minHeight, placement, zIndex, ...props }, ref) => {
-    const contextValue = useMemo(() => ({ isCompact }), [isCompact]);
+  (
+    {
+      button,
+      children,
+      isCompact,
+      focusedValue: _focusedValue,
+      defaultFocusedValue,
+      defaultExpanded,
+      isExpanded: _isExpanded,
+      selectedItems,
+      onChange,
+      ...props
+    },
+    ref
+  ) => {
+    const triggerRef = useRef(null);
+    const menuRef = useRef(null);
+    const items = toItems(children);
+    /* istanbul ignore next */
     const theme = useContext(ThemeContext) || DEFAULT_THEME;
-    const floatingPlacement = toFloatingPlacement(theme.rtl, placement);
+    const environment = useWindow(theme);
+
+    const {
+      isExpanded,
+      focusedValue,
+      getTriggerProps,
+      getMenuProps,
+      getItemProps,
+      getItemGroupProps,
+      getSeparatorProps
+    } = useMenu({
+      rtl: theme.rtl,
+      environment,
+      defaultFocusedValue,
+      focusedValue: _focusedValue,
+      defaultExpanded,
+      isExpanded: _isExpanded,
+      selectedItems,
+      items,
+      menuRef,
+      triggerRef,
+      onChange
+    });
+
+    const triggerProps: IMenuButtonProps = {
+      ...getTriggerProps({ type: 'button' }),
+      ...(isCompact && { size: 'small' }),
```

Comment:
My reasoning for applying sizing here is inspired by Combobox - where `isCompact` changes size of both input and listbox. 

However, I'm not opposed to excluding it from here and requiring a consumer to apply `size` to a custom `button` instance, in favor of a strict trigger/menu prop separation (like all the other props).

Reply:
Makes sense. So in that scenario, a consumer could Choose Your Own Adventure™?

- Set `button` (string) + `buttonProps`; or
- `button` (callback)